app/domain/authentication/authn_gcp/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,88 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of restrictions extraction that are set on a Conjur host or user as annotations.
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        role_class:     ::Role,
+        resource_class: ::Resource,
+        logger:         Rails.logger
+      },
+      inputs:       %i(account username extraction_prefix)
+    ) do
+
+      def call
+        extract_resource_restrictions
+        restrictions_list
```

Comment:
optional name change: `resource_restrictions`.

The class name is `ExtractResourceRestrictions`. seems logical to return `resource_restrictions`.